group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -1105,32 +1106,32 @@ private <IN, OUT> OUT handleOperationException(
             case UNKNOWN_SERVER_ERROR:
                 log.error("Operation {} with {} hit an unexpected exception: {}.",
                     operationName, operationInput, exception.getMessage(), exception);
-                return handler.apply(Errors.UNKNOWN_SERVER_ERROR, null);
+                return handler.apply(Errors.UNKNOWN_SERVER_ERROR, Errors.UNKNOWN_SERVER_ERROR.message());
 
             case NETWORK_EXCEPTION:
                 // When committing offsets transactionally, we now verify the transaction with the
                 // transaction coordinator. Verification can fail with `NETWORK_EXCEPTION`, a
                 // retriable error which older clients may not expect and retry correctly. We
                 // translate the error to `COORDINATOR_LOAD_IN_PROGRESS` because it causes clients
                 // to retry the request without an unnecessary coordinator lookup.
-                return handler.apply(Errors.COORDINATOR_LOAD_IN_PROGRESS, null);
+                return handler.apply(Errors.COORDINATOR_LOAD_IN_PROGRESS, Errors.COORDINATOR_LOAD_IN_PROGRESS.message());
 
             case UNKNOWN_TOPIC_OR_PARTITION:
             case NOT_ENOUGH_REPLICAS:
             case REQUEST_TIMED_OUT:
-                return handler.apply(Errors.COORDINATOR_NOT_AVAILABLE, null);
+                return handler.apply(Errors.COORDINATOR_NOT_AVAILABLE, Errors.COORDINATOR_NOT_AVAILABLE.message());
 
             case NOT_LEADER_OR_FOLLOWER:
             case KAFKA_STORAGE_ERROR:
-                return handler.apply(Errors.NOT_COORDINATOR, null);
+                return handler.apply(Errors.NOT_COORDINATOR, Errors.NOT_COORDINATOR.message());
 
             case MESSAGE_TOO_LARGE:
             case RECORD_LIST_TOO_LARGE:
             case INVALID_FETCH_SIZE:
-                return handler.apply(Errors.UNKNOWN_SERVER_ERROR, null);
+                return handler.apply(Errors.UNKNOWN_SERVER_ERROR, Errors.UNKNOWN_SERVER_ERROR.message());
 
             default:
-                return handler.apply(apiError.error(), apiError.message());
+                return handler.apply(apiError.error(), apiError.messageWithFallback());
```

Comment:
@dajac I'm not sure the original idea about the returned message. What do you think about changing to use a non-null message ?

Reply:
I don't know about the original intention either, but related to my other [comment](https://github.com/apache/kafka/pull/16497/files#r1679707045), it seems to me that we should maybe stick to the `apiError.message()` just to make sure we're not disregarding the logic that the `ApiError` applies to the message (it removes it if it considers it may be sensitive info, or a generic msg that adds no value to the error itself).